src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1285,21 +1285,149 @@ internal bool HasCollectionExpressionApplicableAddMethod(SyntaxNode syntax, Type
             }
         }
 
+        internal bool HasCollectionExpressionAddMethod(SyntaxNode syntax, TypeSymbol targetType)
+        {
+            const string methodName = "Add";
+            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+
+            var implicitReceiver = new BoundObjectOrCollectionValuePlaceholder(syntax, isNewInstance: true, targetType) { WasCompilerGenerated = true };
+            var memberAccess = BindInstanceMemberAccess(
+                syntax,
+                right: syntax,
+                boundLeft: implicitReceiver,
+                rightName: methodName,
+                rightArity: 0,
+                typeArgumentsSyntax: default,
+                typeArgumentsWithAnnotations: default,
+                invoked: true,
+                indexed: false,
+                BindingDiagnosticBag.Discarded);
+            if (memberAccess is not BoundMethodGroup methodGroup)
+            {
+                return false;
+            }
+
+            Debug.Assert(methodGroup.SearchExtensionMethods);
+
+            if (methodGroup.ResultKind == LookupResultKind.Viable &&
+                methodGroup.Methods.Any(m => isApplicableAddMethod(m.GetConstructedLeastOverriddenMethod(accessingTypeOpt: null, requireSameReturnType: false), expectingExtensionMethod: false)))
```

Comment:
>GetConstructedLeastOverriddenMethod [](http://example.com/codeflow?start=69&length=35)

I think `OverloadResolution`  checks ```IsOptional``` on parameters of the method itself, `params` are checked on a least overridden definition.